models/User.js

```diff
@@ -35,117 +35,40 @@ class User {
   }
 
   static create(name, nN, instrument, part, role, spotId, email, password) {
-    return new Promise((resolve, reject) => {
-      const client = utils.db.createClient();
-      const sql = `
-        INSERT INTO users
-        (name, namenumber, instrument, part, role, spotId, email, password)
-        VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
-        `;
-
-      client.connect();
-      client.on('error', reject);
-
-      const query = client.query(sql, [name, nN, instrument, part, role, spotId, email, password]);
-
-      query.on('error', (err) => {
-        client.end();
-        reject(err);
-      });
+    const sql = `
+      INSERT INTO users
+      (name, namenumber, instrument, part, role, spotId, email, password)
+      VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
+      `;
 
-      query.on('end', () => {
-        client.end();
-        resolve();
-      });
-    });
+    return db.query(sql, [name, nN, instrument, part, role, spotId, email, password]);
   }
 
-  canChallengeForPerformance(user, performanceId) {
-    return new Promise((resolve, reject) => {
-      if (!performanceId) {
-        resolve(false);
-      }
-
-      const client = utils.db.createClient();
-      const sql = queries.canChallengeForPerformance;
-
-      client.connect();
-      client.on('error', reject);
-
-      const query = client.query(sql, [user.nameNumber, performanceId, user.spotId]);
-
-      // With this query, no rows means that the user can't challenge, so if the row event
-      // Triggers, that means the user can challenge and we'll resolve true
-      query.on('row', () => {
-        client.end();
-        resolve(true);
-      });
-
-      query.on('end', () => {
-        client.end();
-        resolve(false);
-      });
-
-      query.on('error', (err) => {
-        reject(err);
-        client.end();
-      });
-    });
+  static canChallengeForPerformance(user, performanceId) {
+    if (!performanceId) {
+      return Promise.resolve(false);
```

Comment:
1) Will performanceId ever be 0?
2) If performanceId is not passed, is that an indication of a programming error, or is it something that could happen normally? If the former, that would make this a rejection.

1 may be a ship stopper, 2 almost certainly isn't